Saucelabs integration

alexose opened the pull request:
This pull request integrates Travis + Nightwatch + Saucelab and cleans up some of webpack's configuration options.

There's a bit more to do (generating the production assets could be a little better.  Also I'm not 100% sure if this breaks people's dev environments or not), but this is most of it.

alexose committed:
attempt travis saucelabs integration

alexose committed:
attempt travis saucelabs integration

alexose committed:
update travis yml, ditch old conf files

alexose committed:
update travis yml, ditch old conf files

alexose committed:
add files

alexose committed:
add files

alexose committed:
Merge branch 'master' into saucelabs-integration

alexose committed:
Merge branch 'master' into saucelabs-integration

alexose committed:
add tunnel ID to make saucelabs connect work.  add nightwatch reporting function so saucelabs knows when the tests are finished.

alexose committed:
add tunnel ID to make saucelabs connect work.  add nightwatch reporting function so saucelabs knows when the tests are finished.

alexose committed:
npm start server before integration tests run

alexose committed:
npm start server before integration tests run

alexose committed:
fix npm run command for travis

alexose committed:
fix npm run command for travis

alexose committed:
avoid running prestart

alexose committed:
avoid running prestart

alexose committed:
argh. missing quote

alexose committed:
trying again

alexose committed:
trying again

alexose committed:
add webpack

alexose committed:
add webpack

alexose committed:
v____v

alexose committed:
v____v

alexose committed:
move e2e travis testing commands to script

alexose committed:
move e2e travis testing commands to script

alexose committed:
allow webpack-prod to finish before server boots

alexose committed:
allow webpack-prod to finish before server boots

alexose committed:
symlink prod assets to dev for now

alexose committed:
symlink prod assets to dev for now

alexose committed:
update scripts

alexose committed:
update scripts

alexose committed:
try removing generated/dev dir

alexose committed:
try removing generated/dev dir

alexose committed:
add pwd

alexose committed:
add pwd

alexose committed:
remove special prod config in favor of environment variables

alexose committed:
remove special prod config in favor of environment variables

alexose committed:
fix conflicts in package.json

alexose committed:
remove workaround

alexose committed:
remove workaround

awong-dev commented:
Nifty. My main concern is with the sleep 3.  Would like to get that into something w/o the race condition before submission if it's not too hard.

awong-dev committed:
Add test locking down the golden test data. (#85)

* Add test locking down the golden test data.

This ensures the test data used in the UI is the same as the test data
file used for the API.  It does this by taking each test data blob and
ensuring a test exists that runs it through each validation/conversion
function to check that the exact same result is produced at each stage.

* Make travis run the server tests.

eek.

alexose committed:
Merge branch 'saucelabs-integration' of https://github.com/department-of-veterans-affairs/healthcare-application into saucelabs-integration

Conflicts:
	package.json

alexose committed:
reformat .travis.yml

alexose committed:
fix package.json

alexose committed:
remove sleep statement in favor of something more robust

alexose commented:
GitHub appears to be comparing this branch against an older version of `master`.  I'm not sure why, but I created a new PR, and it seems to have fixed it:

https://github.com/department-of-veterans-affairs/healthcare-application/pull/92
